sklearn/model_selection/tests/test_search.py

```diff
@@ -1199,6 +1200,72 @@ def compare_refit_methods_when_refit_with_acc(search_multi, search_acc, refit):
         assert getattr(search_multi, key) == getattr(search_acc, key)
 
 
+@pytest.mark.filterwarnings("ignore:The parameter 'iid' is deprecated")  # 0.24
+@pytest.mark.parametrize('search_cv', [RandomizedSearchCV, GridSearchCV])
+def test_search_cv_score_samples_error(search_cv):
+    if search_cv == RandomizedSearchCV:
+        clf = search_cv(estimator=DecisionTreeClassifier(),
+                        param_distributions={'max_depth': [5, 10]})
+    else:
+        clf = search_cv(estimator=DecisionTreeClassifier(),
+                        param_grid={'max_depth': [5, 10]})
+
+    X, y = make_blobs(n_samples=100, n_features=4, random_state=42)
+    clf.fit(X, y)
+
+    # Make sure to error out when underlying estimator does not implement
+    # the method `score_samples`
+    err_msg = "AttributeError: 'DecisionTreeClassifier' object has " \
+              "no attribute 'score_samples'"
+    assert_raise_message(AttributeError, err_msg, clf.score_samples, X)
+
+
+@pytest.mark.filterwarnings("ignore:The parameter 'iid' is deprecated")  # 0.24
+@pytest.mark.parametrize('search_cv', [RandomizedSearchCV, GridSearchCV])
+def test_search_cv_score_samples_method(search_cv):
```

Comment:
We can create the objects here:

```py
@pytest.mark.parametrize('search_cv', 
    [RandomizedSearchCV(estimator=..., param_distributions=...)
```

And we will not need the `if` statement in the test.